### PR TITLE
Bump iOS minimum version in podspec to support Xcode 14.3

### PR DESCRIPTION
**Pull request by mdmathias**
Fixes https://github.com/openid/AppAuth-iOS/issues/759

### CODE REVIEW

File: AppAuth.podspec [hunk 1]
```diff
@@ -31,7 +31,7 @@ It follows the OAuth 2.0 for Native Apps best current practice
   #       classes of AppAuth with tokens on watchOS and tvOS, but currently the
   #       library won't help you obtain authorization grants on those platforms.
 
-  s.ios.deployment_target = "7.0"
+  s.ios.deployment_target = "9.0"
   s.osx.deployment_target = "10.9"
```
Comment (paulb777) [inline AppAuth.podspec:35]: Probably need 10.12 here to be consistent with 2015 releases.

Comment (jmagman) [reply AppAuth.podspec:35]: The `libarclite_iphoneos` issue reported in the comment on https://github.com/openid/AppAuth-iOS/issues/759#issuecomment-1487880042 reproduces on macOS when the deployment target is < macOS 10.11 so this should be at least 10.11 to actually fix that linker issue, if not higher.

Comment (mdmathias) [reply AppAuth.podspec]: Thanks for the catch. I have updated this. Podspec is linting locally. Publishing this PRs draft.

**Review by paulb777 — APPROVED** [review]
LGTM. 

Remember to also bump the podspec version to 1.6.1 or 1.7.0 before publishing to CocoaPods trunk. My recommendation without full context would be 1.6.1 to make it as seamless as possible for clients to get the update.

**Comment by mdmathias** [comment]
Thanks @paulb777 for the reminder. The last time we bumped a minimum OS version, we did a minor release.

https://github.com/openid/AppAuth-iOS/commit/de08e7847ec9bbc064eda107399d20b423c0e076

I'm inclined to stick with that precedent. Thoughts?

**Comment by paulb777** [comment]
If the version bump affected Xcodes that are still currently be used, I would agree that it needs to be a minor version update. Since this is changing from a long ago version to a still unsupported version, I think a patch version update is better so that clients that lock to a minor version also get the fix without a respin of their podspec.

**Review by jmagman — APPROVED** [review]
LGTM, thanks!

**Review by ch40w31 — APPROVED** [review]

**Pull request merged**